src/Lockedfield.php

```diff
@@ -68,6 +68,18 @@ public static function canCreate()
         return Session::haveRight(self::$rightname, UPDATE);
     }
 
+    public function canPurgeItem()
+    {
+        $item = new $this->fields['itemtype']();
+        if (
+            $item->getFromDB($this->fields['items_id']) //not a global lock
+            && !Session::haveAccessToEntity($item->getEntityID(), $item->isRecursive()) // no access to main item entity
```

Comment:
If the item is not assigned to an entity (I guess it may happen), this will always return `false`, as `$item->getEntityID()` will return `-1`.

```suggestion
            && $item->isEntityAssign()
            && !Session::haveAccessToEntity($item->getEntityID(), $item->isRecursive()) // no access to main item entity
```